staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -438,15 +478,31 @@ func (p *applyPatcher) applyPatchToCurrentObject(obj runtime.Object) (runtime.Ob
 		return nil, errors.NewBadRequest(fmt.Sprintf("error decoding YAML: %v", err))
 	}
 
-	return p.fieldManager.Apply(obj, patchObj, p.options.FieldManager, force)
+	if p.validationDirective == metav1.FieldValidationStrict {
+		strictPatchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+		if err := yaml.UnmarshalStrict(p.patch, &strictPatchObj.Object); err != nil {
+			return nil, errors.NewBadRequest(fmt.Sprintf("error strict decoding YAML: %v", err))
+		}
+
+	}
+	if p.validationDirective == metav1.FieldValidationWarn {
+		strictPatchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
```

Comment:
is there a way to make use of the decoded object rather than decoding multiple times?